openmdao/core/problem.py

```diff
@@ -220,10 +220,7 @@ def setup(self, check=True, out_stream=sys.stdout):
         # TODO: handle any automatic grouping of systems here...
 
         # divide MPI communicators among subsystems
-        if MPI:
-            self.root._setup_communicators(MPI.COMM_WORLD)
-        else:
-            self.root._setup_communicators(FakeComm())
+        self.root._setup_communicators(self._impl.world_comm())
```

Comment:
When we have nested problem instances I'm sure we'll have to tweak the setup process.  Potentially we could have a comm arg passed to setup that defaults to None, indicating that the current Problem is the top level Problem or something like that.  In any case, all of this will have to be addressed as part of the nested Problem story.